generators/java/generator-utils/src/main/resources/QueryStringMapper.java

```diff
@@ -0,0 +1,91 @@
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import java.util.Iterator;
+import java.util.Map;
+import okhttp3.HttpUrl;
+
+public class QueryStringMapper {
+
+    private static final ObjectMapper MAPPER = ObjectMappers.JSON_MAPPER;
+
+    public static void addQueryParameter(HttpUrl.Builder httpUrl, String key, Object value) {
+        JsonNode nested = MAPPER.valueToTree(value);
+
+        ObjectNode flat;
+
+        if (nested.isObject()) {
+            flat = flattenObject((ObjectNode) nested);
+        } else if (nested.isArray()) {
+            flat = flattenArray((ArrayNode) nested, "");
+        } else {
+            throw new IllegalArgumentException("Expected value to serialize to JSON object or array. Instead, got "
+                    + (value == null ? null : value.toString()));
+        }
```

Comment:
Given that this is a core utility we'll include in the SDK, I'll include a _mega nit_ - I like to place the variable adjacent to the control flow where it's set (i.e. without a newline), so that it's more logically grouped.

```suggestion
        ObjectNode flat;
        if (nested.isObject()) {
            flat = flattenObject((ObjectNode) nested);
        } else if (nested.isArray()) {
            flat = flattenArray((ArrayNode) nested, "");
        } else {
            throw new IllegalArgumentException("Expected value to serialize to JSON object or array. Instead, got "
                    + (value == null ? null : value.toString()));
        }
```